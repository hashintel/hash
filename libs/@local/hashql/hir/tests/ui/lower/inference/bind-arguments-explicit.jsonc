//@ run: fail
//@ description: Ensures type inference fails when explicitly bound generic arguments conflict with provided argument types.
[
  "`+`<Number, Number>",
  //~^ ERROR Type variable has incompatible upper and lower bounds
  { "#literal": 42.12 },
  { "#literal": "42" }
]
