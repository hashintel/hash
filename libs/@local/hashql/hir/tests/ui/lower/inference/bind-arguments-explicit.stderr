error[type-check::bound-constraint-violation]: Type bound constraint violation
  ╭▸ 
1 │ //@ run: fail
  │ ─ Upper bound `Number` is not a supertype of the lower bound
  ‡
4 │   "`+`<Number, Number>",
  │    ━━━━━━━━━━━━━━━━━━━ Type variable has incompatible upper and lower bounds
  ‡
7 │   { "#literal": "42" }
  │                 ──── Lower bound `String` must be a subtype of the upper bound
  │
  ├ help: These type bounds create an impossible constraint. To fix this:
  │       1. Modify `String` to be a proper subtype of `Number`
  │       2. Or adjust `Number` to be a supertype of `String`
  │       3. Or check your code for contradictory type assertions
  ╰ note: During type inference, when a variable has both upper and lower bounds, the relationship 'lower <: upper' must hold. This ensures a valid solution exists in the type system. When this relationship is violated, it means your code is requiring contradictory types for the same variable.